evm/chains/shard.py

```diff
@@ -0,0 +1,69 @@
+from typing import (
+    Union,
+)
+from eth_typing import (
+    Hash32,
+)
+
+from evm.db.shard import (
+    Availability,
+    ShardDB,
+)
+
+from evm.rlp.headers import (
+    CollationHeader,
+)
+from evm.rlp.collations import (
+    Collation,
+)
+
+
+class Shard:
+
+    def __init__(self, shard_db: ShardDB, shard_id: int) -> None:
+        self.shard_id = shard_id
+        self.shard_db = shard_db
+
+    def _check_shard_id(self, header_or_collation: Union[CollationHeader, Collation]) -> None:
+        if header_or_collation.shard_id != self.shard_id:
+            raise ValueError("Header or collation belongs to shard {} instead of shard {}".format(
```

Comment:
`ValidationError` would be more inline with other library conventions.  Also, this could be done as a stand-alone validator function to reduce class size.